include/gunrock/applications/async/util/util.cuh

```diff
@@ -0,0 +1,75 @@
+#ifndef CUDA_COMMON
```

Comment:
Another clean-up step will be adding a C++ interface to all of these to `include/gunrock/cuda/`, I personally do not line `#define`. We can also replace the HEADER `#define`s like `CUDA_COMMON` with `#pragma once`.

Reply:
I'm not sure all of the functions in here are needed right now.

You don't like `#define`'s like this: `#define TID (threadIdx.x+blockIdx.x*blockDim.x)` ? 

Any particular reason?  It's nice to have a shortcut, but does add a layer of indirection in the code.

Reply:
Nothing wrong with shortcuts, but it should be written out with a C++ interface rather than `#defines`.